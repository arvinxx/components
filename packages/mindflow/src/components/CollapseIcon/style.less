@import '../../themes/theme';

.mind-node-collapse {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 20px;
  height: 20px;
  border: 1px solid transparent;
  border-radius: 20px;
  cursor: pointer;

  &-icon {
    font-size: 13px;
  }
}
