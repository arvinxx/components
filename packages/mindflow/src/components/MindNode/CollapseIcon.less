@import '../../themes/theme';

.hover {
  box-shadow: 0 0 8px rgba(0, 0, 0, 0.05);
}

.mind-node {
  &-collapse {
    position: absolute;
    top: 10px;
    right: -10px;
    z-index: 100;
    display: flex;
    align-items: center;
    justify-content: center;
    width: 20px;
    height: 20px;
    border: 1px solid transparent;
    border-radius: 20px;
    cursor: pointer;

    &-icon {
      font-size: 13px;
    }
  }
}
