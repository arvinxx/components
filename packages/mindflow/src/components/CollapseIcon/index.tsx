import type { FC } from 'react';
import { useState } from 'react';
import { MinusOutlined, PlusOutlined } from '@ant-design/icons';
import { useMindflowService } from '../../store/useMindflowContext';
import './style.less';

interface CollapseIconProps {
  id: string;
  collapsed: boolean;
  color: string;
}

const CollapseIcon: FC<CollapseIconProps> = ({ id, collapsed, color }) => {
  const [isHovered, setHovered] = useState(false);
  const { toggleNodeCollapsed } = useMindflowService();

  return (
    <div
      className="mind-node-collapse"
      style={{
        borderColor: color,
        background: isHovered ? color : 'white',
      }}
      onMouseEnter={() => {
        setHovered(true);
      }}
      onClick={() => {
        toggleNodeCollapsed(id);
      }}
    >
      <div
        className="mind-node-collapse-icon"
        style={{ color: isHovered ? '#fff' : color }}
        onMouseEnter={() => {
          setHovered(true);
        }}
        onMouseLeave={() => {
          setHovered(false);
        }}
        onClick={() => {
          console.log('icon');
        }}
      >
        {collapsed ? (
          <PlusOutlined
            onMouseEnter={() => {
              setHovered(true);
            }}
            onClick={() => {
              console.log('svg');
            }}
          />
        ) : (
          <MinusOutlined
            onMouseEnter={() => {
              setHovered(true);
            }}
            onClick={() => {
              console.log('svg');
            }}
          />
        )}
      </div>
    </div>
  );
};
export default CollapseIcon;
