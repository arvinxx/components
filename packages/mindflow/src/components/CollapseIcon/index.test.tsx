import React from 'react';
import { render } from '@testing-library/react';

import CollapseIcon from './index';

describe('CollapseIcon', () => {
  it('正常渲染', () => {
    const { container } = render(
      <CollapseIcon color={'red'} id={'123'} collapsed={false} />,
    );
    expect(container).toMatchSnapshot();
  });
  it('折叠状态', () => {
    const { container } = render(
      <CollapseIcon color={'blue'} id={'123'} collapsed />,
    );
    expect(container).toMatchSnapshot();
  });
});
