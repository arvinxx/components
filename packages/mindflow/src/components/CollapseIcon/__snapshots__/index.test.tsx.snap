// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollapseIcon 折叠状态 1`] = `
<div>
  <div
    class="mind-node-collapse"
    style="border-color: blue; background: white;"
  >
    <div
      class="mind-node-collapse-icon"
      style="color: blue;"
    >
      <span
        aria-label="plus"
        class="anticon anticon-plus"
        role="img"
        tabindex="-1"
      >
        <svg
          aria-hidden="true"
          data-icon="plus"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <defs>
            <style />
          </defs>
          <path
            d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
          />
          <path
            d="M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z"
          />
        </svg>
      </span>
    </div>
  </div>
</div>
`;

exports[`CollapseIcon 正常渲染 1`] = `
<div>
  <div
    class="mind-node-collapse"
    style="border-color: red; background: white;"
  >
    <div
      class="mind-node-collapse-icon"
      style="color: red;"
    >
      <span
        aria-label="minus"
        class="anticon anticon-minus"
        role="img"
        tabindex="-1"
      >
        <svg
          aria-hidden="true"
          data-icon="minus"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M872 474H152c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h720c4.4 0 8-3.6 8-8v-60c0-4.4-3.6-8-8-8z"
          />
        </svg>
      </span>
    </div>
  </div>
</div>
`;
